pkg/cache/v3/linear.go

```diff
@@ -453,33 +587,38 @@ func (cache *LinearCache) CreateWatch(request *Request, sub Subscription, value
 	cache.mu.Lock()
 	defer cache.mu.Unlock()
 
-	response := cache.computeSotwResponse(watch, ignoreCurrentSubscriptionResources)
+	response, err := cache.computeSotwResponse(watch, ignoreCurrentSubscriptionResources)
+	if err != nil {
+		cache.log.Warnf("[linear cache] error computing watch response: %s", err)
+		return nil, fmt.Errorf("failed to compute the watch respnse: %w", err)
```

Comment:
Logging and also returning the error is almost never the right thing to do, since there is high chances it will get logged twice. Here IIUC at least for ADS it will also severe the stream with an gRPC `status.Unknown` error because of https://github.com/envoyproxy/go-control-plane/blob/652ffb49f6ff95ea4c81dc20eaf6d2f36e0ba75d/pkg/server/sotw/v3/ads.go#L115.

The logic is cluttered with error handling that, IIUC, can only happen due to protobuf marshalling error, which I don't know how to make them happen. So maybe a better option would be to log and either swallow or panic directly inside the function that marshalls protobuf. I think for most users of go control plane, the protobuf structs are going to be generated directly inside the system, so there is very little chances that we hit one of those marshalling errors.